arraycontext/container/__init__.py

```diff
@@ -173,6 +173,10 @@ def is_array_container_type(cls: type) -> bool:
         function will say that :class:`numpy.ndarray` is an array container
         type, only object arrays *actually are* array containers.
     """
+    assert isinstance(cls, type), \
+            f"must pass a type, not an instance: '{cls!r}'"
+    assert hasattr(cls, "__mro__"), "'cls' has no attribute '__mro__': "
```

Comment:
Actually, is this `__mro__` check needed? Everything that is an instance of `type` should have an `__mro__` defined, right?

Reply:
I don't think it's needed.

Just to check, `isinstance(..., type)` does seem to successfully rule out the `typing` stuff:
```
>>> from typing import Sequence
>>> isinstance(Sequence, type)
False
```

Reply:
Thanks for checking! Removed in 1a28536.